praxis/layers/models.py

```diff
@@ -136,6 +151,9 @@ def compute_xent_loss_helper(
       fraction_of_correct_next_step_preds=(mean_acc, metric_weight),
       num_predictions=(num_preds, jnp.array(1.0, num_preds.dtype)),
   )
+  if report_strict_acc:
```

Comment:
Could you move the new computation under this conditional too? It feels awkward to do the computation no matter what, but only add it to the metrics if desired.